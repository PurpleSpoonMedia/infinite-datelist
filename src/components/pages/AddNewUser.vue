<template>
  <div>
    <h3 class="text-center">Add new user</h3>
    <app-user-form
      formType="addNewUser"
      @userData="saveUser"
    ></app-user-form>
  </div>
</template>

<script>
import UsersForm from '../UserForm';

export default {
  methods: {
    saveUser(user) {
      this.$store.dispatch('addUser', user);
    }
  },
  components: {
    appUserForm: UsersForm
  }
}
</script>
