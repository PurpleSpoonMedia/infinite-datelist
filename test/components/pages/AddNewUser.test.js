import { shallowMount } from '@vue/test-utils';
import AddNewUser from '../../../src/components/pages/AddNewUser.vue';
import users from '../../fixtures/users';

describe('AddNewUser.vue', () => {
  test('should render AddNewUser correctly', () => {
    const wrapper = shallowMount(AddNewUser);
    expect(wrapper.html()).toMatchSnapshot();
  })

  test('should call saveUser method when userData is emitted', () => {
    const saveUser = jest.fn();
    const wrapper = shallowMount(AddNewUser);

    wrapper.vm.$on('userData', saveUser);
    wrapper.vm.$emit('userData', users[0]);

    expect(saveUser).toHaveBeenLastCalledWith(users[0]);
  })
});
