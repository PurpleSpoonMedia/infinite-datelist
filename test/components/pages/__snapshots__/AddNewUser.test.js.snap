// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewUser.vue should render AddNewUser correctly 1`] = `
<div>
  <h3 class="text-center">Add new user</h3>
  <appuserform-stub userprop="[object Object]" formtype="addNewUser"></appuserform-stub>
</div>
`;
